### DISCUSSION TITLE
IPerf3 UDP test requires TCP ports to be opened in firewall as well

### DISCUSSION THREAD

**Discussion by MightyChee** [esnet/iperf#1559] [Q&A]
Trying with udp test, I deployed iperf in azure vm (windows 10), i had the following networking Inbound port rules and the test failed, Iperf client was unable to connect to Iperf Server.

server command: Iperf3.exe -s -i 1 -p 5201 -f KBytes 

![image](https://github.com/esnet/iperf/assets/19426044/7c58caba-ad23-4598-aeb5-427b60dbbc75)

But as soon as i changed the TCP port setting as below, then the test went ok.

![image](https://github.com/esnet/iperf/assets/19426044/019242f8-a951-4a4c-8ecd-93fa88bafa69)

Anyone know why UDP test need the same UDP port number to be opened in TCP port setting as well?

Thank you

**Comment by davidBar-On** [comment]
Hi, the control interface between the server and client  is always over TCP, even if the test itself is UDP.

**Reply by MightyChee** [reply]
Hi, thank you for replying. 

Is that mean Iperf works like Twamp which it has control phase (tcp) and test phase (udp/tcp depending on the protocol chosen)?

**Comment by davidBar-On** [comment]
> Is that mean Iperf works like Twamp which it has control phase (tcp) and test phase (udp/tcp depending on the protocol chosen)?

I am not familiar with Twmap, but by briefly reading [this explanation](https://www.juniper.net/documentation/us/en/software/junos/flow-monitoring/topics/concept/twamp-overview.html) it seems that the answer is yes, when only two entities are used as shown in Figure 2.  The iperf3-control and the TWAMP-control interfaces have similar usages.  Note that these interfaces are used throughout the test and not only to start a test - the Twamp description say "TWAMP-Control  - Initiates, starts, **and ends** test sessions".
